### ISSUE TITLE
Add multiple columns at once to the query builder

### ISSUE THREAD

**Issue by samaguire** [DaxStudio/DaxStudio#1131]
**Is your feature request related to a problem? Please describe.**
As per the title. It's frustrating to add them one by one.

**Describe the solution you'd like**
Use 'ctrl' to select multiple columns and/or measures from the metadata well and drag them all to the query builder panes and have them all added at once.

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by dgosbell** [comment]
Unfortunately, the TreeView control does not natively support ctrl to multi-select nodes. I could maybe build my own control, but that would be a fair amount of work. But I'm also thinking that I could override the double click on columns and measures to send them to the columns area of the Query Builder as I'm guessing that it's probably adding a list of columns to there that is the major issue. 

Would that help here?

Another option is to use a combination of searching or clicking with the keyboard short cuts https://daxstudio.org/docs/features/query-builder/#keyboard-shortcuts to quickly add items to the Query Builder

**Comment by samaguire** [comment]
Yes, that's right. I'm trying to add columns to the columns area in query builder.

I don't think changing the default double click behaviour is a good idea. I often build the core of the code I'm after using the query editor, then edit it in the DAX editor pane - using the double click.

Ah, shortcut keys. Not the first thing I think about. Are you able to update the water mark to have some text that to highlight the shortcut keys? This would serv as a reminder.

![image](https://github.com/DaxStudio/DaxStudio/assets/62320770/a0bfa4f5-7c24-4c7c-b488-b50455c37d8d)

**Comment by dgosbell** [comment]
> I don't think changing the default double click behaviour is a good idea. I often build the core of the code I'm after using the query editor, then edit it in the DAX editor pane - using the double click.

So my idea here is to make this dynamic, so if the Query Builder is open the columns go there, and once you close the query builder the current behaviour kicks back in. But I could make that configurable in the options.

> Are you able to update the water mark to have some text that to highlight the shortcut keys?

Yes, I'm sure I can do something there. I currently have a tooltip, but it's only triggering when you are over the watermark icon. I've fixed that so it works for the whole drop area, but I'll see about adding a link to the shortcut docs

![image](https://github.com/DaxStudio/DaxStudio/assets/8241830/6cad4147-9cbd-41b9-8cd3-d898148d06e8)

**Comment by samaguire** [comment]
Having the double click as an option would be awesome 👍

Ah. I didn't know there was a tooltip. I'd hold off on the text since you've changed the on hover to the whole panel. I think that'll do the job. 👍

Regards
Stephen

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
